comp110-worksheet-6/OxoBoard.cs

```diff
@@ -10,41 +10,120 @@ public enum Mark { None, O, X };
+        /// </summary>
+        /// <param name="width"></param>
+        /// <param name="height"></param>
+        /// <param name="inARow"></param>
+        public OxoBoard(int width = 3, int height = 3, int inARow = 3)
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
+            gameBoard = new Mark[width, height];
 		}
 
 		// Return the contents of the specified square.
 		public Mark GetSquare(int x, int y)
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
-		}
+            //getting the square that the player is inputing.
+            return gameBoard[x, y];
+        }
 
-		// If the specified square is currently empty, fill it with mark and return true.
-		// If the square is not empty, leave it as-is and return False.
-		public bool SetSquare(int x, int y, Mark mark)
-		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
+        /// <summary>
+        ///  If the specified square is currently empty, fill it with mark and return true
+        ///  If the square is not empty, leave it as-is and return False.
+        /// </summary>
+        /// <param name="x"></param>
+        /// <param name="y"></param>
+        /// <param name="mark"></param>
+        /// <returns></returns>
+        public bool SetSquare(int x, int y, Mark mark)
+		{	
+
+            if(gameBoard[x,y] == Mark.None)
+            {
+                gameBoard[x, y] = mark;
+                return true;
+            }
+            return false;
 		}
 
 		// If there are still empty squares on the board, return false.
 		// If there are no empty squares, return true.
 		public bool IsBoardFull()
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
+            for (int x = 0; x < gameBoard.GetLength(0); x++)
+            {
+                for (int y = 0; y < gameBoard.GetLength(1); y++)
+                {
+                    if(gameBoard[x,y] == Mark.None)
+                    {
+                        return false;
+                    }
+                }
+            }
+            return true;
 		}
 
-		// If a player has three in a row, return Mark.O or Mark.X depending on which player.
-		// Otherwise, return Mark.None.
+		/// <summary>
+        /// Comparing all the filled spaces to eachother to check if there is a three in a row of any kind.
+        /// If there is then it will return the starting tile to select the winner, if there is no winner then
+        /// it will mark none.
+        /// </summary>
+        /// <returns></returns>
 		public Mark GetWinner()
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
-		}
+            if(gameBoard[0,0] == gameBoard[1,0] && gameBoard[0,0] == gameBoard[2,0])
```

Comment:
You also need to check that `gameBoard[0,0] != Mark.None` -- otherwise your checks will detect a line of three empty squares